materialize-motherduck/staged_file.go

```diff
@@ -0,0 +1,213 @@
+package main
```

Comment:
This is pretty much a copy/paste from redshift, with a few changes - namely redshift uses a manifest file, whereas Duckdb uses a list of file names.